File: Examples/test-suite/perl5/cpp_namespace_runme.pl
use strict;
use warnings;
use Test::More tests => 12;
BEGIN { use_ok('cpp_namespace') }
require_ok('cpp_namespace');

my $n = cpp_namespace::fact(4);
is($n, 24, "return value");

is($cpp_namespace::Foo, 42, "variable value");

my $t = cpp_namespace::Test->new();
is($t->method(), "Test::method", "method return value");
is(cpp_namespace::do_method($t), "Test::method", "return value");

is(cpp_namespace::do_method2($t), "Test::method", "return value");

cpp_namespace::weird("hello", 4);

undef $t;

my $t2 = cpp_namespace::Test2->new();
my $t3 = cpp_namespace::Test3->new();
my $t4 = cpp_namespace::Test4->new();
my $t5 = cpp_namespace::Test5->new();

is(cpp_namespace::foo3(42), 42, 'return value');
is(cpp_namespace::do_method3($t2, 40), "Test2::method", "return value");
is(cpp_namespace::do_method3($t3, 40), "Test3::method", "return value");
is(cpp_namespace::do_method3($t4, 40), "Test4::method", "return value");
is(cpp_namespace::do_method3($t5, 40), "Test5::method", "return value");

File: Examples/test-suite/lua/cpp_namespace_runme.lua
require("import")
require("cpp_namespace")
cn=cpp_namespace
catch_undef_globs() -- catch "undefined" global variables

assert(cn.fact(4) == 24)
assert(cn.Foo == 42)

t1 = cn.Test()
assert(t1:method() == "Test::method")

cn.weird("t1", 4)

assert(cn.do_method(t1) == "Test::method")
assert(cn.do_method2(t1) == "Test::method")

t2 = cn.Test2()
assert(t2:method() == "Test2::method")


assert(cn.foo3(5) == 5)

assert(cn.do_method3(t2, 7) == "Test2::method")
